### PR TITLE
refactor(exporters/elasticsearch): don't re-serialize to add sequence

### PR DESCRIPTION
**Pull request by lenaschoenburg**
To prevent having to re-serialize records for adding the `sequence` field, we configure a Jackson mix-in that automatically adds the `sequence` field.

### CODE REVIEW

File: exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/BulkIndexRequest.java [hunk 1]
```diff
@@ -74,16 +73,14 @@ void index(
...
-    enhancedRecord.put(RECORD_SEQUENCE_PROPERTY, recordSequence.sequence());
-
-    return MAPPER.writer().writeValueAsBytes(enhancedRecord);
+    return MAPPER
+        .writer()
+        // Enhance the serialized record by its sequence number. The sequence number is not a part
+        // of the record itself but a special property for Elasticsearch. It can be used to limit
+        // the number of records when reading from the index, for example, by using a range query.
+        // Read https://github.com/camunda/zeebe/issues/10568 for details.
+        .withAttribute(RECORD_SEQUENCE_PROPERTY, recordSequence.sequence())
```
Comment (saig0) [inline exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/BulkIndexRequest.java:82]: :+1: Wow. It's magic. :magic_wand: 

I was looking for something like that but I wasn't aware of this feature. It's a perfect fit here. :dart:

**Review by saig0 — APPROVED** [review]
@oleschoenburg just awesome. :1st_place_medal: 

Thank you for making this code much more efficient. :seedling:

**Comment by saig0** [comment]
bors merge

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**